tests/testlib/s2n_test_server_client.c

```diff
@@ -77,8 +77,15 @@ S2N_RESULT s2n_negotiate_test_server_and_client_with_early_data(struct s2n_conne
     bool server_early_done = false, client_early_done = false;
     do {
         if (!client_early_done) {
+            /* Adding 0 to a null value is undefined behavior.
+             * This prevents undefined behavior in the case where the offset is 0.
+             */
+            uint8_t *next_early_data_to_send = early_data_to_send->data;
+            if (total_data_sent != 0) {
+                next_early_data_to_send += total_data_sent;
+            }
```

Comment:
For all your comments, I wonder if they're too intrusive. This is a small issue unrelated to any of the code around it, and you've got big verbose multi-line comments. Maybe you should just do:
```suggestion
            uint8_t *next_early_data_to_send = early_data_to_send->data;
            /* Adding 0 to a NULL value is undefined behavior */
            if (total_data_sent != 0) {
                next_early_data_to_send += total_data_sent;
            }
```